cmd/metalctl/app/move.go

```diff
@@ -0,0 +1,347 @@
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	sourceKubeconfig string
+	targetKubeconfig string
+	crdsOnly         bool
+	crsOnly          bool
+	namespace        string
+	dryRun           bool
+	verbose          bool
+)
+
+func NewMoveCommand() *cobra.Command {
+	move := &cobra.Command{
+		Use:   "move",
+		Short: "Move metal-operator CRDs and CRs from one cluster to another",
+		RunE:  runMove,
+	}
+	move.Flags().StringVar(&sourceKubeconfig, "source-kubeconfig", "", "Kubeconfig pointing to the source cluster")
+	move.Flags().StringVar(&targetKubeconfig, "target-kubeconfig", "", "Kubeconfig pointing to the target cluster")
+	move.Flags().BoolVar(&crdsOnly, "crds-only", false, "migrate only the CRDs without CRs")
+	move.Flags().BoolVar(&crsOnly, "crs-only", false, "migrate only the CRs without CRDs")
+	move.Flags().StringVar(&namespace, "namespace", "", "namespace to filter CRDs and CRs to migrate. Defaults to all namespaces if not specified")
+	move.Flags().BoolVar(&dryRun, "dry-run", false, "show what would be moved without executing the migration")
+	move.Flags().BoolVar(&verbose, "verbose", false, "enable verbose logging for detailed output during migration")
+	move.MarkFlagRequired("source-kubeconfig")
+	move.MarkFlagRequired("target-kubeconfig")
+
+	if verbose {
+		slog.SetLogLoggerLevel(slog.LevelDebug)
+	}
+	return move
+}
+
+type Clients struct {
+	source client.Client
+	target client.Client
+}
+
+func makeClient(kubeconfig string) (client.Client, error) {
+	cfg, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load cluster kubeconfig: %w", err)
+	}
+	return client.New(cfg, client.Options{Scheme: scheme})
+}
+
+func makeClients() (Clients, error) {
+	var clients Clients
+	var err error
+
+	clients.source, err = makeClient(sourceKubeconfig)
+	if err != nil {
+		return clients, fmt.Errorf("failed to construct a source cluster client: %w", err)
+	}
+	clients.target, err = makeClient(targetKubeconfig)
+	if err != nil {
+		return clients, fmt.Errorf("failed to construct a target cluster client: %w", err)
+	}
+	return clients, nil
+}
+
+func getMetalCrds(ctx context.Context, cl client.Client) ([]*apiextensionsv1.CustomResourceDefinition, error) {
+	crds := &apiextensionsv1.CustomResourceDefinitionList{}
+	if err := cl.List(ctx, crds); err != nil {
+		return nil, fmt.Errorf("couldn't list CRDs: %w", err)
+	}
+
+	metalCrds := make([]*apiextensionsv1.CustomResourceDefinition, 0)
+	for _, crd := range crds.Items {
+		if crd.Spec.Group == metalv1alphav1.GroupVersion.Group {
+			metalCrds = append(metalCrds, &crd)
+		}
+	}
+
+	return metalCrds, nil
+}
+
+func getMetalCrs(ctx context.Context, cl client.Client, crds []*apiextensionsv1.CustomResourceDefinition) ([]*unstructured.Unstructured, error) {
+	crs := make([]*unstructured.Unstructured, 0)
```

Comment:
Why not using a `[]client.Object` here and work with the concrete types? I would avoid using `unstructured` whenever possible. As `metalctl` is part of the `metal-operator` the type import should not be a problem.

Reply:
I used `unstructured` because I have a hard time implementing a loop over each CRD. If I want to do something like:
```
for _, CRList := range[]client.ObjectList{&metalv1alphav1.BMCList{}, &metalv1alphav1.BMCSecretList{}, ...} {
    client.List(CRList)
```
I can't collect its items as `client.ObjectList` is an interface (see https://github.com/kubernetes-sigs/controller-runtime/issues/2697).

So without some nasty casting I should list all CRs one by one and collect its items:
```
bmcList := &metalv1alphav1.BMCList{}
client.List(bmcList)
for _, cr := range bmcList.Items {
	crs = append(crs, &cr)
}

bmcSecretList := &metalv1alphav1.BMCSecretList{}
client.List(bmcSecretList)
for _, cr := range bmcSecretList.Items {
	crs = append(crs, &cr)
}
...
```

This is the solution you want me to use or there is a more clever way to do it?